cli/lib/util.js

```diff
@@ -193,9 +193,35 @@ const util = {
     .mapValues((value) => { // stringify to 1 or 0
       return value ? '1' : '0'
     })
+    .extend(util.getNodeOptions(options))
     .value()
   },
 
+  getNodeOptions (options, nodeVersion) {
+    if (!nodeVersion) {
+      nodeVersion = Number(process.versions.node.split('.')[0])
+    }
+
+    if (options.dev && nodeVersion < 12) {
+      // `node` is used when --dev is passed, so this won't work if Node is too old
+      logger.warn('(dev-mode warning only) NODE_OPTIONS=--max-http-header-size could not be set. See https://github.com/cypress-io/cypress/pull/5452')
+
```

Comment:
Will the new max-header log in DEBUG logs for all of our users? Cause I'd like any new options to be logged in their DEBUG logs for future debugging.

Reply:
It is not logged anywhere :/ But, it should always be passed, as this is the only condition where it wouldn't be.